### PR TITLE
fix: Added a format to the timestamp value in v2 logging method to en…

### LINKED ISSUE #59: Update timestamp value in message object to have a consistent format
https://github.com/avioconsulting/mule-custom-logger/blob/8a4cb55608abc5168a4286200fc28378e8f08f24/src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java#L82

The line of code above does not provide a specific format for the output for the timestamp values. Because of this there can be different formats based on the underlying jdk of what the app is running on. This is especially proving important when trying to parse messages in things like DataDog. 

The proposed consistent format would be: `"yyyy-MM-dd'T'HH:mm:ss.SSSZ"`

### PR DESCRIPTION
**Pull request by tannersherman**
Added a format to the timestamp value in v2 logging method to ensure format is always the same.
 
 Closes #59

### CODE REVIEW

File: src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java [hunk 1]
```diff
@@ -7,10 +7,9 @@
 import com.avioconsulting.mule.logger.api.processor.MessageAttributes;
 import com.avioconsulting.mule.logger.internal.config.CustomLoggerConfiguration;
 import com.avioconsulting.mule.logger.internal.utils.CustomLoggerUtils;
+import java.text.SimpleDateFormat;
 import java.time.Instant;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (adesjardin) [inline src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java:12]: In general we should avoid wildcard imports and keep specific classes

File: src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java [hunk 2]
```diff
@@ -79,7 +78,9 @@ public void log(LogProperties logProperties,
       Object oTelContext = messageAttributes.getOTelContextObject();
 
       Map<String, Object> logContext = new LinkedHashMap<>();
-      logContext.put("timestamp", Instant.now().toString());
+      SimpleDateFormat timestampFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+      timestampFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
```
[the remark this answers is not in the document]
Comment (adesjardin) [reply src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java:82]: Also the DateTimeFormatter object should be static and only initialized, along with the time zone lookup.  There is overhead to those things we don't want to do every time we write a log.

**Review by adesjardin — CHANGES_REQUESTED** [review]

File: src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tannersherman) [reply src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java]: Thanks for the feedback, updated to reflect these notes

File: src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java (hunk 1 above)
Comment (tannersherman) [reply src/main/java/com/avioconsulting/mule/logger/internal/CustomLogger.java]: I think intellij automatically did this bc we had a growing number of java.util imports. But fixed this as well

**Review by manikmagar — APPROVED** [review]

**Review by adesjardin — APPROVED** [review]

**Pull request merged**